Refactoring regarding meter history broke proper notification with INotifyProperyChange

christoh opened the pull request.

HugoX104447 committed:
Refactor HVAC service to not depend on application settings

HugoX104447 committed:
Introduce library to host common code shared by FroniusMonitor and FroniusPhone

Move IDataCollectionService, DataCollectionService, HomeAutomationSystem,
SolarDataEventArgs, SmartMeterCalibrationHistoryItem and SettingsBase into
a shared libray that hosts common code for both applications. All code is
really tied to the application that is build on top of the Fronius library,
nothing that can be reused per se by a different home automation system or
a different setup of devices.

# Conflicts:
#	FroniusMonitor.sln

christoh committed:
Merge pull request #21 from HugoX104447/refactor-homeautomation

WIP: Refactor IDataCollectionService, DataCollectionService and HomeA…

HugoX104447 committed:
Rework some code, follow the "Use Async All the Way" pattern

* Use async Main function, since there is no return value anyway.
* Use availabe async methods if possible.
* Do not synchronously block the thread in main.

HugoX104447 committed:
Refactor settings loading into generic methods in SettingsBase

Unifies behavior across the applications, updating check-sums, clearing
passwords etc. Still, gives enough flexibility to extend lodaing or saving
settings if there is still a need. Use SemaphoreSlim to protect resource
access instead of busy-waiting (polling with Task.Delay etc.).

christoh committed:
Merge pull request #25 from HugoX104447/refactor-settings

Refactor settings loading into generic methods in SettingsBase

HugoX104447 committed:
Amend commit cfddb57, build fix, minor cleanup

* Re-add line I accidentally removed.
* Remove redundant type argument specification.
* Fix FroniusPhone build, use async method temporarily.

christoh committed:
Merge pull request #27 from HugoX104447/refactor-settings

Amend commit cfddb57, I accidentally removed one line too many

christoh committed:
Merge branch 'dev' into async-main

christoh committed:
Merge pull request #26 from HugoX104447/async-main

Rework some code, follow the "Use Async All the Way" pattern

christoh commented:
Refactoring regarding meter history broke proper notification with INotifyProperyChange
![image](https://github.com/christoh/FroniusMonitor/assets/17645550/320809a7-3ec8-449d-98d7-6c26db7e8fd4)

HugoX104447 commented:
I must have broken this in commit [6f7e694d19dfcd33733a7831d59cad700ebfa256](https://github.com/christoh/FroniusMonitor/pull/21/commits/6f7e694d19dfcd33733a7831d59cad700ebfa256), will look into it ASAP

HugoX104447 committed:
Amend commit 6f7e69, fix broken house consumption watts

Before commit 6f7e69, notifications about Load and or Grid power corrected
values was send while setting the either of these properties in Gen24PowerFlow.
Now we will send the update notification right after setting the SitePowerFlow
property.

christoh committed:
Merge pull request #29 from HugoX104447/bug-fix

Bug fix
